ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -412,7 +422,8 @@ static enum pkcs11_rc create_storage_attributes(struct obj_attrs **out,
 		return rc;
 
 	return set_optional_attributes(out, temp, pkcs11_any_object_optional,
-				       ARRAY_SIZE(pkcs11_any_object_optional));
+				       ARRAY_SIZE(pkcs11_any_object_optional),
+				       true);
```

Comment:
To prevent later confusion, i think a wrapper helper would be better:

```c
static enum pkcs11_rc _set_optional_attributes(struct obj_attrs **out,
					       struct obj_attrs *temp,
					       uint32_t const *bp,
					       size_t bp_count,
					       bool default_to_null)
{
 ...
}

static enum pkcs11_rc set_optional_attribute_or_null(struct obj_attrs **out,
						     struct obj_attrs *temp,
						     uint32_t const *bp,
						     size_t bp_count)
{
	return set_optional_attributes_or_null(..., true);
}

static enum pkcs11_rc set_optional_attributes(struct obj_attrs **out,
					      struct obj_attrs *temp,
					      uint32_t const *bp,
					      size_t bp_count)
{
	return set_optional_attributes_or_null(..., false);
}

Reply:
ok..I  will create the wrappers